core/src/main/java/io/narayana/nta/logparsing/common/Filter.java

```
@@ -22,13 +22,15 @@
 
 package io.narayana.nta.logparsing.common;
 
+import java.io.IOException;
+
 /**
  * @Author Alex Creasy &lt;a.r.creasy@newcastle.ac.uk$gt;
  * Date: 19/07/2013
  * Time: 21:11
  */
 public interface Filter {
 
-    boolean matches(String line);
+    boolean matches(String line) throws IOException,IndexOutOfBoundsException;
```

Why it need to throw these tow Exception ?